### DISCUSSION TITLE
Godot 4 code completion: sort offered class members by inheritance

### DISCUSSION THREAD

**Discussion by Exzerios** [godotengine/godot-proposals#4153] [Editor]
Topic name pretty much describes it. For now code completion feature doesn't help in cases when you begin to forget function names because of large codebase, and you have to peak into those files to get it.

Working with some "Object -> Node -> Block -> TickableBlock -> Furnace" when writing furnace_obj.something we usually need something from the Furnace class itself, either it's unique class member or overriden one, and very rarely something derived from Object. So ordering completion options beginning with current class and then up the inheritance tree could speed up things a little. Would be especially handy considering Godot 4 typing improvements. And, while not being any critical, seems to be pretty easy to implement.

**Comment by Mickeon** [comment]
Very major peeve. It's incredibly annoying to make use of the hints, only to be overwhelmed by Object constants first and foremost, that you're highly likely not going to use because your class extends something way beyond the basic Object and it would be ideal to see what its specialised properties and methods are, first.

**Comment by Calinou** [comment]
There's already proposals on the subject: https://github.com/godotengine/godot-proposals/issues/99 and https://github.com/godotengine/godot-proposals/issues/4189

**Comment by HolonProduction** [comment]
This is what the current sorting algorithm is doing, so I'll be closing this discussion
